src/jit/hwintrinsicxarch.cpp

```diff
@@ -234,6 +234,57 @@ int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic)
     return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
 }
 
+//------------------------------------------------------------------------
+// numArgsOfHWIntrinsic: get the number of arguments based on table and
+// if numArgs is -1 check number of arguments using GenTreeHWIntrinsic node
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function
+//    node      -- GenTreeHWIntrinsic* node
+//
+// Return Value:
+//     number of arguments
+//
+int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic, GenTreeHWIntrinsic* node)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+
+    int numArgs = hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
+    if (numArgs != -1)
+    {
+        return numArgs;
+    }
+
+    assert(numArgs == -1);
```

Comment:
this assert is pointless.

A better assert would be to assert `numArgs >= 0` before returning `numArgs` above, or to change the check to `if (numArgs >= 0) { return numArgs; }` and then to keep this assert (`numArgs == -1`).